File: Code/SimpleBudget.API/Models/ProductPrices/ProductPriceEditModel.cs
using System.ComponentModel.DataAnnotations;

namespace SimpleBudget.API.Models
{
    public class ProductPriceEditModel
    {
        [Required]
        public string ProductName { get; set; } = default!;

        public string? CompanyName { get; set; } = default!;

        public string? CategoryName { get; set; }

        [Required]
        public string PriceDate { get; set; } = default!;

        [Required]
        public decimal Price { get; set; }

        [Required]
        public bool IsDiscount { get; set; }

        public int? Quantity { get; set; }

        public string? Description { get; set; }
    }
}

File: Code/client/src/api/models/ProductPriceEditModel.ts
export interface ProductPriceEditModel {
    productName: string;
    companyName: string | undefined | null;
    categoryName: string | undefined | null;
    priceDate: string;
    price: number;
    isDiscount: boolean;
    quantity: number | undefined | null;
    description: string | undefined | null;
}